delta-sharing/core/src/kernel.rs

```diff
@@ -114,4 +115,117 @@ impl TableQueryHandler for KernelQueryHandler {
+
+        let protocol = snapshot.protocol();
+        let metadata = snapshot.metadata();
+        let response_format = request
+            .response_format
+            .first()
+            .and_then(|format| format.parse::<ResponseFormat>().ok())
+            .unwrap_or(ResponseFormat::Parquet);
+
+        let response = match response_format {
+            ResponseFormat::Parquet => {
+                let parquet_protocol_action = t::ParquetWrappedAction {
+                    action: Some(t::parquet_wrapped_action::Action::ParquetProtocol(
+                        ParquetWrappedProtocol {
+                            min_reader_version: protocol.min_reader_version,
+                        },
+                    )),
+                };
+                let parquet_metadata_action = t::ParquetWrappedAction {
+                    action: Some(t::parquet_wrapped_action::Action::ParquetMetadata(
+                        ParquetWrappedMetadata {
+                            id: metadata.id.clone(),
+                            name: metadata.name.clone(),
+                            description: metadata.description.clone(),
+                            schema_string: metadata.schema_string.clone(),
+                            partition_columns: metadata.partition_columns.clone(),
+                            configuration: metadata.configuration.clone(),
+                            format: Some(t::ParquetFormat {
+                                provider: metadata.format.provider.clone(),
+                            }),
+                            version: None,
+                            size: None,
+                            num_files: None,
+                        },
+                    )),
+                };
+
+                let actions = t::ParquetWrappedResponse {
+                    actions: vec![parquet_protocol_action, parquet_metadata_action],
+                };
+                t::get_table_metadata_response::Response::Parquet(actions)
+            }
+            ResponseFormat::Delta => {
+                let delta_protocol_action = t::DeltaWrappedAction {
+                    action: Some(t::delta_wrapped_action::Action::DeltaProtocol(
+                        t::DeltaWrappedProtocol {
+                            delta_protocol: Some(t::DeltaProtocol {
+                                min_reader_version: protocol.min_reader_version,
+                                min_writer_version: protocol.min_writer_version,
+                                reader_features: protocol
+                                    .reader_features
+                                    .clone()
+                                    .unwrap_or_default(),
+                                writer_features: protocol
+                                    .writer_features
+                                    .clone()
+                                    .unwrap_or_default(),
+                            }),
```

Comment:
Should we define `From` implementations for the `Protocol` and `Metadata` types we get from kernel in the core crate?.